contracts/Tokenator.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.24;
+
+import {ERC20, ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract Tokenator is ERC20Burnable, Ownable {
+    using SafeERC20 for IERC20;
+
+    IERC20 underlyingToken;
```

Comment:
let's make it public, so people could see what it is

Reply:
done